src/cassandra/future.rs

```diff
@@ -120,11 +122,10 @@ impl Completable for PreparedStatement {
 }
 
 /// A Cassandra future is a normal Rust future.
-impl<T: Completable> futures::Future for CassFuture<T> {
-    type Item = T;
-    type Error = Error;
+impl<T: Completable> Future for CassFuture<T> {
+    type Output = Result<T>;
 
-    fn poll(&mut self) -> futures::Poll<Self::Item, Self::Error> {
+    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
```

Comment:
I haven't done any futures migrations myself, but I have skimmed https://www.ncameron.org/blog/migrating-a-crate-from-futures-0-1-to-0-3/ . Can you explain to me why this pin is sufficient and correct?

Reply:
I'm not sure I understand the question exactly. Switching from `&mut self` -> `self: Pin<&mut Self>` is how one updates manual future implementations (it is a mechanical change).

> why this pin is sufficient and correct?

Why would it not be sufficient and correct? In theory, since `CassFuture` is `!Unpin` there should be no need for `unsafe`.

Reply:
As long as all types are `Unpin` (which they should be in a mechanical upgrade from 0.1 -> std::future), then changing `&mut self` -> `self: Pin<&mut Self>` is purely a mechanical change. When types are `Unpin`, `Pin<&mut Self>` is roughly identical to `&mut self`. Things get complicated once you work with `!Unpin` types, i.e. futures returned from `async fn`.

Reply:
As long as we are not messing with `!Unpin`, using `Pin` is equivalent to `&mut self`, so this is as safe as it was before :D

Reply:
In order to make sure that no one is accidentally messing with `!Unpin`, I've also gone ahead and scoped down the `unsafe` block to wrap the minimal amount of code (the CFFI calls).

Reply:
OK, thanks - that makes sense.